rosbag2_to_video/api.py

```diff
@@ -58,17 +78,20 @@ def convert_bag_to_video(args):
     if not bag_reader.has_next():
         print('empty bag file')
         return
-    msg_type = get_message('sensor_msgs/msg/Image')
+    msg_type = get_message(type_map[args.topic])
 
     _, data, _ = bag_reader.read_next()
     image_msg = deserialize_message(data, msg_type)
-    # cv2.VideoWriter expects BGR encoding 
-    cv_image = cvbridge.imgmsg_to_cv2(image_msg, 'bgr8')
+    # cv2.VideoWriter expects BGR encoding
+    if type_map[args.topic] == 'sensor_msgs/msg/Image':
+        cv_image = cvbridge.imgmsg_to_cv2(image_msg, 'bgr8')
+    elif type_map[args.topic] == 'sensor_msgs/msg/CompressedImage':
+        cv_image = cvbridge.compressed_imgmsg_to_cv2(image_msg, 'bgr8')
     height, width, _ = cv_image.shape
     success = video_writer.open(
         args.output,
         cv2.CAP_FFMPEG,
-        cv2.VideoWriter.fourcc('a', 'v', 'c', '1'),
+        cv2.VideoWriter.fourcc('m', 'p', '4', 'v'),
```

Comment:
please don't change the default encoding, H264 is a good default.
If you want to add an option to select another encoding, that's fine

Reply:
I added an argument for video codec. The default value is avc1. https://github.com/ivanpauno/rosbag2_to_video/pull/1/commits/595932065e54c9ac92b164b08a54ab8f4b51b195

Reply:
I found out that avc1 and h264 is not supported in opencv python shipped by pip or apt due to the license issue.
https://stackoverflow.com/questions/59290569/opencv-video-writer-unable-to-find-codec-or-avc1
https://github.com/opencv/opencv-python/issues/207